### PR TITLE
Setup start and stop endpoints

### LINKED ISSUE #60: Setup start/stop missions endpoints
**Describe the new feature you would like to see**
The backend should provide the frontend with endpoints for starting and stopping robots/missions.

**Describe the solution you'd like**
Endpoints that forwards the request to start a robot with a given mission to the appropriate ISAR endpoint. Similarly for stop commands.

**How will this feature affect the current Threat Model?**
The potential to create harm with these endpoints are greater than the others. Different scope/authorization for making calls to these endpoints should be considered.

### PR DESCRIPTION
**Pull request by vetlek**
Closes #60

### CODE REVIEW

File: backend/src/flotilla/services/isar/requests.py
Comment (nicholasdalhaug) [inline backend/src/flotilla/services/isar/requests.py:20]: Sounds like a list of requests. Is this not just an API? Also a typo in "interface"
```suggestion
class IsarAPI(IsarAPIInterface):
```

File: backend/src/flotilla/services/request_handler.py
Comment (nicholasdalhaug) [inline backend/src/flotilla/services/request_handler.py:69]: I see we already use `**kwargs`, why not just use that for all?
```suggestion
        **kwargs,
```

File: backend/tests/mocks/isar_requests_mock.py
Comment (nicholasdalhaug) [inline backend/tests/mocks/isar_requests_mock.py:41]:
```suggestion
        return self.default_start_mission_response
```

File: backend/tests/mocks/isar_requests_mock.py
Comment (nicholasdalhaug) [inline backend/tests/mocks/isar_requests_mock.py:52]: I do not see why we need this functionality in the mock. Is it not enough to mock `start` and `stop`?

**Review by nicholasdalhaug — CHANGES_REQUESTED** [review]
Please tell me if you think I misunderstand something 😁

File: backend/src/flotilla/api/missions_api.py
Comment (vetlek) [reply backend/src/flotilla/api/missions_api.py:62]: This is for documentation purposes (So the different responses shows up in the Swagger). Agree that the error messages should be the same though.

File: backend/src/flotilla/api/robots_api.py
Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:41]: I think the names should correspond to CRUD operations and the get_* should be reserved for the API-function.

File: backend/src/flotilla/api/missions_api.py
Comment (aestene) [reply backend/src/flotilla/api/missions_api.py:62]: I agree, the same goes for the other endpoint above. There should be a way to only have the string change in one location.

File: backend/src/flotilla/settings/settings.py
Comment (vetlek) [reply backend/src/flotilla/settings/settings.py:17]: The default is used if the variable is not in the .env file, so it is not a case of two different defaults.

File: backend/src/flotilla/api/missions_api.py
Comment (aestene) [inline backend/src/flotilla/api/missions_api.py:77]: Instead of writing status codes by writing the int directly an enum should be used. This goes for both in code and documentation decorators.

Use this library: https://docs.python.org/3/library/http.html
image

File: backend/src/flotilla/services/request_handler.py
Comment (vetlek) [reply backend/src/flotilla/services/request_handler.py:69]: It is nice to see the different available parameters when you want to use the function.

File: backend/src/flotilla/api/robots_api.py
Comment (aestene) [inline backend/src/flotilla/api/robots_api.py:83]: Should be sorted according to status number.

File: backend/src/flotilla/api/robots_api.py
Comment (aestene) [reply backend/src/flotilla/api/robots_api.py:101]: Yes, rather throw an exception in `read_robot_by_id`

File: backend/tests/mocks/isar_requests_mock.py
Comment (vetlek) [reply backend/tests/mocks/isar_requests_mock.py:52]: So you can be able to add your custom responses if you want to test how the api response to a particular response from ISAR

File: backend/tests/mocks/isar_requests_mock.py
Comment (aestene) [inline backend/tests/mocks/isar_requests_mock.py:36]:
```suggestion
        self.default_start_mission_response: Response = default_start_mission_response()
        self.default_stop_response: Response = default_stop_response()
```

File: backend/src/flotilla/api/robots_api.py
Comment (aestene) [inline backend/src/flotilla/api/robots_api.py:118]: This should not be necessary. If you look at the `RequestHandler` it uses a function `response.raise_for_status()`. This will raise an error for any status code outside the 200 range and as such we should not get to this if statement as it will have been raised as a `RequestException` in `RequestHandler`

**Review by aestene — CHANGES_REQUESTED** [review]
See comments

File: backend/src/flotilla/api/robots_api.py
Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:118]: Should we change this behavior? Seems like we can loose a lot of useful information from using the raise_from_status, like if there is a conflict. Don't know if we would handle it any different, but might want to pass the information about the particular issue?

File: backend/src/flotilla/services/isar/requests.py
Comment (vetlek) [reply backend/src/flotilla/services/isar/requests.py:40]: Handled since we raise error when status code is not in 200 range

File: backend/src/flotilla/api/robots_api.py
Comment (Christdej) [inline backend/src/flotilla/api/robots_api.py:83]: Remove all responses except 200, we dont need this at the moment

File: backend/src/flotilla/api/robots_api.py
Comment (Christdej) [inline backend/src/flotilla/api/robots_api.py:144]: Remove all responses except 200, we dont need this at the moment

**Review by Christdej — CHANGES_REQUESTED** [review]

File: backend/src/flotilla/api/robots_api.py
Comment (aestene) [reply backend/src/flotilla/api/robots_api.py:118]: The error message itself is kept as we use `raise error from e`. As such we should only need to log the error to have it available internally. Typically you would like to not send all the error information to the frontend but rather log it in the backend itself. As such I think this should be fine as we are saving the information in the error itself.
https://github.com/equinor/flotilla/pull/63/files#diff-c5827550d464dc4af6f269546e3cd1aae61565d50740f7837449c79f12ab8436R54-R58

File: backend/src/flotilla/api/robots_api.py
Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:101]: Leave this to #74 since this needs to be done other places that is not in the scope of this PR.

File: backend/src/flotilla/api/robots_api.py
Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:41]: The naming issue is fixed temporarily in this file for now. Added an issue about renaming for all database models #76

File: backend/src/flotilla/api/robots_api.py
Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:118]: Think the try catch around the raise_for_status call should be removed and let the error be handled in the controller.

Comment (vetlek) [reply backend/src/flotilla/api/robots_api.py:118]: But this should probably be done in a separate issue.

Comment (aestene) [reply backend/src/flotilla/api/robots_api.py]: Sure, that may be more correct for this case.

**Review by aestene — APPROVED** [review]
LGTM. Check with @nicholasdalhaug and @Christdej if they are happy with the review before merging.

File: backend/src/flotilla/settings/settings.py
Comment (nicholasdalhaug) [reply backend/src/flotilla/settings/settings.py:17]: I do not see why we need all these environment variables, but I can ask that in an other setting.

File: backend/tests/test_api/test_missions_api.py
Comment (nicholasdalhaug) [inline backend/tests/test_api/test_missions_api.py:63]: The first numbers here are pretty magical, where are they from?

**Review by nicholasdalhaug — CHANGES_REQUESTED** [review]
Still one comment from before that is still not answered: [link](https://github.com/equinor/flotilla/pull/63#discussion_r800747191)

File: backend/src/flotilla/services/isar/requests.py
Comment (vetlek) [reply backend/src/flotilla/services/isar/requests.py]: Hmm, don't know if feel like API is any better

**Review by nicholasdalhaug — APPROVED** [review]

**Pull request merged**
